src/pandablocks/hdf.py

```diff
@@ -101,7 +102,7 @@ def create_dataset(self, field: FieldCapture, raw: bool):
             # No processor, datatype passed through
             dtype = field.type
         return self.hdf_file.create_dataset(
-            f"/{field.name}.{field.capture}",
+            f"/{self.dataset_names.get(field.name, field.name)}.{field.capture}",
```

Comment:
We need a bit more logic here, as per https://github.com/bluesky/ophyd-async/issues/330#issuecomment-2126825286

> If it has a single value then give it the scientific name
> If it is "Min Max" then expose scientific_name + "-min" and scientific_name + "-max"
> If it is "Min Max Mean" then expose scientific_name + "-min", scientific_name + "-max" and scientific_name

Reply:
I've modified this

https://github.com/PandABlocks/PandABlocks-client/blob/ea83a6a01890ea3211a1da5de0888c9c21247cc7/src/pandablocks/hdf.py#L108-L115

The ioc now stores a dict of `record -> field.capture -> dataset_name` (updated whenever the capture mode or dataset name is changed. It's `deepcopy'd` so the filewriter will use the same values at start and finish:

https://github.com/PandABlocks/PandABlocks-ioc/blob/5b0bba8c2863c0caab348af0b53fdfaefb2ea988/src/pandablocks_ioc/ioc.py#L875-L893